paper.md

```diff
@@ -0,0 +1,160 @@
+or Fortran, e,g., deal.II [@dealii2019design] or PETSc [@petsc-efficient]. These languages
+offer high computational performance, but often at the cost of being complex to learn and
+maintain. On the other hand, languages like Python, which are more amenable for rapid
+prototyping or less experienced users, are usually not fast enought to use without
+specialized packages that utilize kernels written in a compiled language. For example,
+Python's well-known NumPy library [@2020NumPy-Array] has its performance-critical code
+implemented in C.
+
+The Julia programming language [@Bezanson_Julia_A_fresh_2017] aims to provide a new approach
+to scientific computing. It strives to solve the two-language problem by providing an
+accessible, high-level syntax together with fast, just-in-time-compiled execution
+[@churavy2022bridging]. Due to its native ability to call C or Fortran functions, in
+multi-language projects Julia often acts as a glue code between newly developed
+implementations written in Julia and existing libraries written in C/Fortran.
+
+While there exist other numerical simulation codes in Julia, e.g., Gridap.jl [@Badia2020] or
+Ferrite.jl [@Carlsson_Ferrite_jl], none of them provide the ability to use them directly
+from another programming language. With `libtrixi`, we therefore enable new workflows by
+allowing scientists to connect established research codes to a modern numerical simulation
+package written in Julia. That is, a main program written in C/C++ or Fortran is able to
+execute a simulation set up in Julia with Trixi.jl without sacrifices in performance.
+
+This control direction, where a Julia library is managed from C/Fortran, has been rarely
+used in practice so far. Besides making Trixi.jl available to a wider scientific audience,
+`libtrixi` is thus a research project to investigate the efficacy of using Julia-based
+libraries in existing code environments, and can eventually serve as a blueprint for
+other similar efforts. Questions such as how to retain the flexibility of Julia while
+providing a traditional, fixed API, how to use system-local third-party libraries, or how to
+interact with Fortran are investigated and answered.
+
+
+# Technical overview
+`Libtrixi` consists of three main parts:
+
+- the Julia package
+  [LibTrixi.jl](https://github.com/trixi-framework/libtrixi/tree/main/LibTrixi.jl), which
+  provides a traditional library API to Trixi.jl in Julia,
+- a C API that exposes this Julia API as an ordinary, shared C library,
+- Fortran bindings for the C API.
+
+\autoref{fig:libtrixi_overview} illustrates the general workflow: A main program written in
+C/C++/Fortran interacts with the public-facing C or Fortran API of `libtrixi`. The
+Fortran API is little more than a set of language bindings, with some extra code to handle
+the conversion between Fortran strings and C strings. The C API is slightly more involved,
```

Comment:
It also handles "logical functions" (?)
So write more generally "data types" or just use strings as an example
(I'm fine with with every variant)

Reply:
Good catch! Fixed in 2488874.